include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -1735,6 +1735,67 @@ namespace internal
           output += out_stride;
         }
     }
+  }; /**
+      * Helper class to interpolate values and gradients from cell to face
+      * quadrature points.
+      */
+  template <int dim, int n_points_1d, int n_components, typename Number>
+  struct FEFaceNormalQuadratureEvaluationImpl
+  {
+    template <bool do_evaluate, bool add_into_output>
+    static void
+    interpolate(const internal::MatrixFreeFunctions::ShapeInfo<Number> &data,
+                const Number *                                          input,
+                Number *                                                output,
+                const bool         do_gradients,
+                const unsigned int face_no)
+    {
+      AssertDimension(n_points_1d, data.data.front().quadrature.size());
```

Comment:
Since this function is almost the same as the other `FEFaceNormalEvaluationImpl`, I wonder whether we can unify the calls and let both function call an internal function `interpolate_generic`. What I'm thinking about is to let `interpolate` and `interpolate_quadrature` extract the data array (`shape_data_on_face[face_no % 2]` or `quadrature_data_on_face[face_no % 2]`) as well as the `in_stride` and `out_stride` and let the `interpolate_generic` (or whatever name we choose) do the job.

Reply:
Good idea. What I was thinking about as a follow-up PR is to make the function dimension independent since I want to use it in 6D. What do you think?

Reply:
Sounds good.